tests/org.eclipse.emf.mwe2.language.tests/src/org/eclipse/emf/mwe2/language/tests/validation/Mwe2ValidatorTest.java

```diff
@@ -39,222 +30,193 @@
+		validationTestHelper.assertWarning(barModule, DECLARED_PROPERTY, UNUSED_LOCAL, "The var 'foo' is never read locally.");
+	}
+
+	@Test public void testDeprecatedElement() {
+		assertWarning(
+				"module m\r\n" + 
+				"\r\n" + 
+				"import org.eclipse.xtext.xtext.generator.XtextGenerator\r\n" + 
+				"import org.eclipse.xtext.xtext.generator.StandardLanguage\r\n" + 
+				"\r\n" + 
+				"Workflow {\r\n" + 
+				"	component = XtextGenerator {\r\n" + 
+				"		language = StandardLanguage {\r\n" + 
+				"			newProjectWizardForEclipse = {}\r\n" + 
+				"		}\r\n" + 
+				"	}\r\n" + 
+				"}", "newProjectWizardForEclipse", ASSIGNMENT, DEPRECATED_ELEMENT,
+				"The 'org.eclipse.xtext.xtext.generator.StandardLanguage.setNewProjectWizardForEclipse' is deprecated.");
+	}
+
+	private void assertWarning(String text, String errorProneText, EClass objectType, String code, String message) {
+		validationTestHelper.assertWarning(parse(text), objectType, code, text.indexOf(errorProneText),
+				errorProneText.length(), message);
+	}
+
+	private void assertError(String text, String errorProneText, EClass objectType, String code, String message) {
+		validationTestHelper.assertError(parse(text), objectType, code, text.indexOf(errorProneText),
+				errorProneText.length(), message);
 	}
 
-	@Test public void testDeprecatedElement() throws Exception {
-		String textModel = 
-			"module m\r\n" + 
-			"\r\n" + 
-			"import org.eclipse.xtext.xtext.generator.XtextGenerator\r\n" + 
-			"import org.eclipse.xtext.xtext.generator.StandardLanguage\r\n" + 
-			"\r\n" + 
-			"Workflow {\r\n" + 
-			"	component = XtextGenerator {\r\n" + 
-			"		language = StandardLanguage {\r\n" + 
-			"			newProjectWizardForEclipse = {}\r\n" + 
-			"		}\r\n" + 
-			"	}\r\n" + 
-			"}";
-		EObject model = getModel(textModel);
-		List<Issue> issues = validate(model);
-		assertEquals(issues.toString(), 1, issues.size());
-		Issue issue = issues.get(0);
-		assertEquals(Mwe2Validator.DEPRECATED_ELEMENT, issue.getCode());
-		assertEquals(Severity.WARNING, issue.getSeverity());
-		assertEquals("The 'org.eclipse.xtext.xtext.generator.StandardLanguage.setNewProjectWizardForEclipse' is deprecated.", issue.getMessage());
+	private void assertError(String text, int errorProneTextOffset, int errorProneTextLength, EClass objectType,
+			String code, String message) {
+		validationTestHelper.assertError(parse(text), objectType, code, errorProneTextOffset, errorProneTextLength, message);
 	}
 
-	private List<Issue> validate(EObject model) {
-		XtextResource res = ((XtextResource)model.eResource());
-		List<Issue> list = res.getResourceServiceProvider().getResourceValidator().validate(res, CheckMode.ALL, CancelIndicator.NullImpl);
-		return list;
+	private void assertNoIssues(String text) {
+		validationTestHelper.assertNoIssues(parse(text));
 	}
 
+	private Module parse(String text) {
```

Comment:
What is this? A method in a test that has a printStacktrace, forgets the original exception and triggers a fail instead of simply declaring `throws Exception`?

Reply:
I created an other PR https://github.com/eclipse/mwe/pull/128 to improve the exception handling as desired.